src/js/modules/filter.js

```diff
@@ -258,12 +258,12 @@ Filter.prototype.generateHeaderFilterElement = function(column, initialValue){
 
 				if (
 					!(
-						(column.definition.headerFilter === 'autocomplete' ||
-							column.definition.editor === 'autocomplete' ||
-							column.definition.headerFilter === 'tickCross' ||
-							column.definition.editor === 'tickCross') &&
-						column.definition.headerFilter === true
```

Comment:
Hi @olifolkerd 

Here what happening is we **applying and (&) conditions with all of or (||) conditions** (**not wrapping the and conditions so would not affect all or (||) conditions**) which returns false in case of 'autocomplete' and 'tickCross' but it should be true.

Because of this `column.definition.headerFilter === true` condition which returns false  in case of 'autocomplete' and 'tickCross' and we are doing and condtion with all of the or conditions and the applying not (!)